components/core/src/clp/regex_utils/ErrorCode.cpp

```diff
@@ -1,59 +1,61 @@
-#include "RegexErrorCode.hpp"
+#include "ErrorCode.hpp"
 
 #include <string>
 
-using clp::regex_utils::RegexErrorCategory;
-using clp::regex_utils::RegexErrorCodeEnum;
+#include <ystdlib/error_handling/ErrorCode.hpp>
+
+using clp::regex_utils::ErrorCodeEnum;
+using RegexErrorCategory = ystdlib::error_handling::ErrorCategory<ErrorCodeEnum>;
```

Comment:
It feels a bit weird that only `RegexErrorCategory` has the `Regex` prefix now. Could we also drop the prefix? Alternatively, you could just remove the using statement since it only shows up twice.